### PR TITLE
missing space for null terminator on baseDir

### PR DESCRIPTION
**Pull request by Scottj1s**

### CODE REVIEW

File: dev/MRTCore/mrt/Core/src/MRM.cpp [hunk 1]
```diff
@@ -993,7 +993,9 @@ STDAPI MrmGetFilePathFromName(_In_opt_ PCWSTR filename, _Outptr_ PWSTR* filePath
         if (SUCCEEDED(wil::TryGetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_BASE_DIRECTORY", baseDir)) && baseDir)
         {
             path.swap(baseDir);
-            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &bufferCount));
+            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &length));
+            size_t bufferCount;
+            RETURN_IF_FAILED(SizeTAdd(length, 1, &bufferCount));
```
Comment (codendone) [inline dev/MRTCore/mrt/Core/src/MRM.cpp:998]: >bufferCount [](http://example.com/codeflow?start=50&length=11)

Adding to this throwaway inner variable?

File: dev/MRTCore/mrt/Core/src/MRM.cpp [hunk 2]
```diff
@@ -993,7 +993,9 @@ STDAPI MrmGetFilePathFromName(_In_opt_ PCWSTR filename, _Outptr_ PWSTR* filePath
         if (SUCCEEDED(wil::TryGetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_BASE_DIRECTORY", baseDir)) && baseDir)
         {
             path.swap(baseDir);
-            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &bufferCount));
+            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &length));
+            size_t bufferCount;
```
[the remark this answers is not in the document]
Comment (JesseCol) [reply dev/MRTCore/mrt/Core/src/MRM.cpp:997]: I must be missing something -- seems like we calculate length+1, and then throw away the result?  Maybe you mean to put the result in bufferCount that's in the outer scope...

[the remark this answers is not in the document]
Comment (Scottj1s) [reply dev/MRTCore/mrt/Core/src/MRM.cpp]: good catch, thx

File: dev/MRTCore/mrt/Core/src/MRM.cpp (hunk 1 above)
Comment (Scottj1s) [reply dev/MRTCore/mrt/Core/src/MRM.cpp:997]: fixed

File: dev/MRTCore/mrt/Core/src/MRM.cpp [hunk 3]
```diff
@@ -993,7 +993,8 @@ STDAPI MrmGetFilePathFromName(_In_opt_ PCWSTR filename, _Outptr_ PWSTR* filePath
         if (SUCCEEDED(wil::TryGetEnvironmentVariableW(L"MICROSOFT_WINDOWSAPPRUNTIME_BASE_DIRECTORY", baseDir)) && baseDir)
         {
             path.swap(baseDir);
-            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &bufferCount));
+            RETURN_IF_FAILED(StringCchLengthW(path.get(), STRSAFE_MAX_CCH, &length));
+            RETURN_IF_FAILED(SizeTAdd(length, 1, &bufferCount));
```
[the remark this answers is not in the document]
Comment (Scottj1s) [reply dev/MRTCore/mrt/Core/src/MRM.cpp:997]: yes, to be consistent with the calculations on lines 980 and 983

**Review by codendone — APPROVED** [review]
:shipit:

**Pull request merged**